app/controllers/api/v1/bulk_download_controller.rb

```diff
@@ -0,0 +1,299 @@
+            key :required, false
+          end
+          response 200 do
+            key :description, 'Curl configuration file with signed URLs for requested data'
+            key :type, :string
+          end
+          response 400 do
+            key :description, 'Invalid study accessions or requested file types'
+          end
+          response 401 do
+            key :description, ApiBaseController.unauthorized
+          end
+          response 403 do
+            key :description, ApiBaseController.forbidden('download with provided auth_token, or download exceeds user quota')
+            schema do
+              key :title, 'Error'
+              property :message do
+                key :type, :string
+                key :description, 'Error message'
+              end
+            end
+          end
+          response 406 do
+            key :description, ApiBaseController.not_acceptable
+          end
+        end
+      end
+
+      def generate_curl_config
+        valid_accessions = []
+        file_ids = []
+        # sanitize study accessions and file types
+        if params[:file_ids]
+          begin
+            file_ids = RequestUtils.validate_id_list(params[:file_ids])
+          rescue ArgumentError
+            render json: {error: 'file_ids must be comma-delimited list of 24-character UUIDs'}, status: 400 and return
+          end
+          matched_study_ids = StudyFile.where(:id.in => file_ids).pluck(:study_id)
+          valid_accessions = Study.where(:id.in => matched_study_ids).pluck(:accession)
+        else
+          valid_accessions = self.class.find_matching_accessions(params[:accessions])
+          sanitized_file_types = self.class.find_matching_file_types(params[:file_types])
+        end
+        check_accession_permissions(valid_accessions)
+
+        # if this is a single-study download, allow for DirectoryListing downloads
+        if valid_accessions.size == 1 && params[:directory].present?
+          study_accession = valid_accessions.first
+          directories = self.class.find_matching_directories(params[:directory], study_accession)
+          directory_files = ::BulkDownloadService.get_requested_directory_files(directories)
+        else
+          directories = []
+          directory_files = []
+        end
+
+        files_requested = nil
+        byebug
+        # get requested files
+        # reference BulkDownloadService as ::BulkDownloadService to avoid NameError when resolving reference
+        if params[:file_ids]
+          files_requested = StudyFile.where(:id.in => file_ids)
+        else
+          files_requested = ::BulkDownloadService.get_requested_files(file_types: sanitized_file_types,
+                                                                      study_accessions: valid_accessions)
+        end
+
+        # determine quota impact & update user's download quota
+        # will throw a RuntimeError if the download exceeds the user's daily quota
```

Comment:
I know that's what the current functionality is (and I wrote it), but for the life of me I can't figure out _why_ we decided that throwing a `RuntimeError` was the right call here...